lib/AST/GenericSignatureBuilder.cpp

```diff
@@ -5092,14 +5095,21 @@ void GenericSignatureBuilder::enumerateRequirements(llvm::function_ref<
     // If we have a superclass, produce a superclass requirement
     if (equivClass->superclass && !equivClass->recursiveSuperclassType) {
       f(RequirementKind::Superclass, archetype, equivClass->superclass,
-        getBestConstraintSource<Type>(equivClass->superclassConstraints));
+        getBestConstraintSource<Type>(equivClass->superclassConstraints,
+                        [&](const Type &type) {
+                          return type->isEqual(equivClass->superclass) ||
+                            equivClass->superclass->hasError();
```

Comment:
We may not have an exact match if there are errors. I've decided that I don't like this approach and will improve on this a bit.